app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java

```diff
@@ -0,0 +1,130 @@
+package uy.com.bix.app.smsproject.controllers;
+
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+
+import android.preference.PreferenceManager;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.Toast;
+
+import net.danlew.android.joda.JodaTimeAndroid;
+
+import org.joda.time.DateTime;
+
+import uy.com.bix.app.smsproject.classes.AlertReceiver;
+
+public class AlarmController extends AppCompatActivity {
+
+	private static AlarmController instance = null;
+	int expirationYear, expirationMonth, expirationDay, expirationHour, expirationMinute, maxMessages;
+	long dateOfNextFiring;
+	String phoneNumber, textMessage;
+	boolean notifyWhenSending, isLastDay, isActive;
+
+	private AlarmController() {}
+
+	public static AlarmController getInstance() {
+		if (instance == null) {
+			instance = new AlarmController();
+		}
+		return instance;
+	}
+
+	public void prepareAndSendMessage (Context context) {
+		SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(context);
+		JodaTimeAndroid.init(context);
+		isActive = settings.getBoolean("Active", false);
+
+		// We make sure the app is active to send messages
+		if (isActive) {
+
+			// We get all the information the user set
+			notifyWhenSending = settings.getBoolean("Notify", false);
```

Comment:
Take in mind for the future, I would create a function for this block and other for 57-62. 

> Functions only have to do one thing